src/callstack.cr

```diff
@@ -165,7 +176,32 @@ struct CallStack
         function = "???"
       end
 
-      "0x#{ip.address.to_s(16)}: #{function} at #{file_line_column}"
+      # We ignore these because they are part of `raise`'s internals,
+      # and we can't rely on a correct filename being available
+      # (for example if on Mac and without running `dsymutil`)
+      #
+      # We also ignore `main` because it's always at the same place
+      # and adds no info.
+      if function.starts_with?("*raise<") ||
+         function.starts_with?("*CallStack::") ||
+         function.starts_with?("*CallStack#") ||
+         function == "main"
```

Comment:
If I have a custom method called `main` it would get hidden from the stacktrace? Just rename `foo` to `main` in your example.

Reply:
You are right. I'll just keep those entries then.

Reply:
I did it because there's only one "main", I don't know why when there's debug info `Foo#main` just appears as `main`. But well, it's not a big deal not removing those entries.

Reply:
Couldn't you just remove the last entry? This would require to count the size, but I don't think this should be a big concern.

Reply:
No, because en release some traces might not be there. I think it's simpler to just keep things as they are now regarding this.

Reply:
Regardless if it might not show up in some scenarios, anything after `__crystal_main` shouldn't be of interest so I think it should be possible to break when this function is processed.
Though it's not a big deal having a line more at the end.